client/packages/requisitions/src/RequestRequisition/DetailView/Footer/StatusChangeButton.tsx

```diff
@@ -109,11 +112,38 @@ const useStatusChangeButton = () => {
   const onConfirmStatusChange = async () => {
     if (!selectedOption) return null;
     try {
-      await update({
+      const result = await update({
+        id,
         status: selectedOption.value,
         comment: getUpdatedComment(),
       });
-      success(t('messages.saved'))();
+
+      if (result.__typename === 'RequisitionNode') {
+        success(t('messages.saved'))();
+      }
+
+      if (result.__typename === 'UpdateRequestRequisitionError') {
+        const { error: updateError } = result;
+
+        switch (updateError.__typename) {
+          case 'OrderingTooManyItems':
+            return error(
+              t('error.ordering-too-many-items', {
+                maxItems: updateError.maxItemsInEmergencyOrder,
+              })
+            )();
+          case 'OtherPartyNotASupplier':
+            return error(t('error.other-party-not-a-supplier'))();
+          case 'CannotEditRequisition':
+            return error(t('error.cannot-edit-requisition'))();
+          case 'OtherPartyNotVisible':
+            return error(t('error.other-party-not-visible'))();
+          case 'RecordNotFound':
+            return error(t('messages.record-not-found'))();
```

Comment:
I've used switch case cause it's a lot cleaner than if else statements... but there are a lot of structured errors we're not covering... Should of these seem like they can just be standard errors... Swap them over?